gtsfm/loader/olsson_loader.py

```diff
@@ -0,0 +1,154 @@
+"""Simple loader class that reads Carl Olsson's dataset from a folder on disk.
+
+Authors: John Lambert
+"""
+
+import glob
+import os
+from pathlib import Path
+from typing import Optional
+
+import numpy as np
+from gtsam import Cal3Bundler, Pose3, Rot3
+from scipy.io import loadmat
+
+import gtsfm.utils.io as io_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+
+class OlssonLoader(LoaderBase):
+    """Simple loader class that reads Carl Olsson's dataset from a folder on disk.
+
+    Ref: http://www.maths.lth.se/matematiklth/personal/calle/dataset/dataset.html
+
+    Folder layout structure:
+    - RGB Images: images/
+    - Intrinsics + Extrinsics data (optional): data.mat
+
+    If explicit intrinsics are not provided, the exif data will be used.
+    """
+
+    def __init__(self, folder: str, image_extension: str = "jpg", use_gt_intrinsics: bool = True) -> None:
+        """Initializes to load from a specified folder on disk
+
+        Args:
+            folder: the base folder for a given scene
+            image_extension: (optional)extension for the image files. Defaults to 'jpg'.
```

Comment:
I dont think we mention (optional) in the args section, do we? It's evident from the function signature.

Reply:
Thanks, fixed